# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/experimental/TypedData.h
# Commit: 6880cf8c51bc
# Full Hash: 6880cf8c51bc665b19b3d14b9fe4edb7960f527e
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-08-13 09:27:46
# Regressor Bug: 1720422
# File Overlap Count: 1
# Description:
#   Bug 1720422 - Switch DOM bindings typed arrays to using JSAPI templates r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119858
# ==============================================================================

diff -r 23cc649c6e14 -r 6880cf8c51bc js/public/experimental/TypedData.h
--- a/js/public/experimental/TypedData.h	Thu Aug 12 20:44:18 2021 +0000
+++ b/js/public/experimental/TypedData.h	Thu Aug 12 20:44:18 2021 +0000
@@ -299,6 +299,11 @@
 // ArrayBufferObject or some sort of ArrayBufferViewObject (see below).
 // Note that this will always hold an unwrapped object.
 class JS_PUBLIC_API ArrayBufferOrView {
+ public:
+  // Typed Arrays will set this to their specific element type.
+  // Everything else just claims to expose things as uint8_t*.
+  using DataType = uint8_t;
+
  protected:
   Heap<JSObject*> obj;
 
